audio-worklet/bitcrusher.js

```diff
@@ -0,0 +1,234 @@
+    this.reduction = options.reduction;
+    this.node_ = this.context_.createScriptProcessor(options.buffersize, options.inputChannels, options.outputChannels);//
+
+    // bind the AudioNode onaudioproess method to the function defined below 
+    this.node_.onaudioprocess = this.onaudioprocess_.bind(this); 
+  }
+  
+  /**
+    Bit crush upon receiving input audio signal, applying two signal distortion effects to the output buffer:
+    - output has reduced bit depth controlled by [options.bits] bits, thereby increasing sample error
+    - the sample changes only every [options.reduction] samples, simulating a sample rate reduction
+    @param {!AudioProcessingEvent} event 
+  */
+  onaudioprocess_ (event) {
+    for (let i = 0; i < event.inputBuffer.numberOfChannels; i++) {
+      Bitcrusher.bitcrush(this.reduction, this.bitDepth, event.inputBuffer.getChannelData(i), event.outputBuffer.getChannelData(i));
+    }
+  }
+
+  /**
+    Reduce the information in given inputBuffer with specified factor and precision,
+    placing results in oututBuffer.
+    @param {number} factor to reduce by
+    @param {number} precision or bit depth of post-processed samples
+    @param {Float32Array} inputBuffer pre-processed buffer
+    @param {Float32Array} outputBuffer post-processed buffer
+  */
+  static bitcrush(factor, precision, inputBuffer, outputBuffer) {
+    if (factor <= 0) factor = 1; // handle erroneous factor input by not sample reducing
+    
+    // for each sample in the input buffer, add new bit crushed sample in outputBuffer
+    let last;
+    for (let j = 0; j < inputBuffer.length; j++) {
+      if (j % factor == 0) { // always activates at iteration 0 since 0 % x == 0
+        last = Bitcrusher.round(inputBuffer[j], precision, 2); // round sample at specified resolution
+      }
+      outputBuffer[j] = last;
+    }
+  }
+
+  /** 
+    Round at the specified unit of precision. For bitcrushing, this reduces the information of a given sample 
+    @param {number} number to be rounded 
+    @param {number} precision of the number
+    @param {number} base of precision
+  */
+  static round(number, precision, base) {
+    let factor = Math.pow(base, precision);
+    let temp = number * factor;
+    let rounded = Math.round(temp);
+    return rounded / factor;
+  };
+  
+  /**
+    Connect this script processor node to target Audio Node
+    @param {AudioNode} target  
+    @return {Bitcrusher} this for chaining
+  */
+  connectOut(dest) {
+    this.node_.connect(dest) 
+    return this;
+  }
+
+  /** 
+    Connect input AudioNode to internal script processor node
+    @param {AudioNode}  
+    @return {Bitcrusher} this for chaining
+  */
+  connectIn(input) {
+    input.connect(this.node_);
+    return this; 
+  }
+  
+  /** 
+    Disconnect input AudioNode to internal script processor node
+    @param {AudioNode} dest to disconnect from
+    @return {Bitcrusher} this for chaining
+  */
+  disconnect (dest) {
```

Comment:
If this is intended to disconnect your bitcrusher from a downstream node, you need to allow more options to match AudioNode.disconnect(), which allows you specify what to disconnect.